Codecademy/gamut
# feat(List): add breakpoint prop and variants

### Overview



Update the List component to take a breakpoint prop instead of only the default "xs" breakpoint
Also add a compact variant for the `spacing` prop

PortalApp PR: https://github.com/codecademy-engineering/portal-app/pull/2381



### PR Checklist

- [x] Related to designs: [Figma](https://www.figma.com/file/Ym03UHdFa0ctNFVoMndKze/Plan-Split%3A-pricing-and-payment-v2?node-id=949%3A71671)
- [x] Related to JIRA ticket: GRW-195

#### Testing Instructions



1. Pass the prop `breakpoint='md'` to the `List` component
2. Resize the screen and observe the wrapping behavior happening at `md` instead of the usual `xs`
An example of this can be seen in [this PR](https://github.com/codecademy-engineering/portal-app/pull/2381) 

![features_responsive1](https://user-images.githubusercontent.com/1364447/185653576-6a52da23-45b4-48ef-9ba8-411e1617ef7b.gif)


#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](http://www.google.fr/ 'Named link title')  | [Monolith Env](http://www.google.fr/ 'Named link title') |
| Portal       | [Portal Link](http://www.google.fr/ 'Named link title')  | [Portal Env](http://www.google.fr/ 'Named link title')   |
| Another Repo | [Another Link](http://www.google.fr/ 'Named link title') | [Another Env](http://www.google.fr/ 'Named link title')  |
